src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,101 @@
+        user.setAge(20);
+        user.setLogin(Integer.toString(user.hashCode()));
+        user.setPassword(Integer.toString(12345678));
+        login = user.getLogin();
+        actual = registrationService.register(user);
+        Assertions.assertNotNull(actual, "Method must return user");
+        Assertions.assertEquals(user, storageDao.get(login), "User must be added to data");
+    }
+
+    @Test
+    void registrationWhenAgeLessThan18_NotOk() {
+        user.setAge(17);
+        user.setLogin(Integer.toString(user.hashCode()));
+        user.setPassword(Integer.toString(12345678));
+        login = user.getLogin();
+        actual = registrationService.register(user);
+        Assertions.assertNull(actual, "Method must return null, "
+                + "invalid age");
+        Assertions.assertNull(storageDao.get(login), "User mustn't be added to data, "
+                + "invalid age");
+    }
+
+    @Test
+    void registrationWithSuchLogin_NotOk() {
+        user.setAge(20);
+        user.setLogin(Integer.toString(user.hashCode()));
+        user.setPassword(Integer.toString(12345678));
+        login = user.getLogin();
+        registrationService.register(user);
+        actual = registrationService.register(user);
+        Assertions.assertNull(actual, "Method must return null, "
+                + "user with a such login is already registered");
+    }
+
+    @Test
+    void registrationWithInvalidPassword_NotOk() {
+        user.setAge(20);
+        user.setLogin(Integer.toString(user.hashCode()));
+        user.setPassword(Integer.toString(123));
+        login = user.getLogin();
+        actual = registrationService.register(user);
+        Assertions.assertNull(actual, "Method must return null, "
+                + "invalid password");
+        Assertions.assertNull(storageDao.get(login), "User mustn't be added to data, "
+                + "invalid password");
+    }
+
+    @Test
+    void registrationWithNullPassword_NotOk() {
+        user.setAge(20);
+        user.setLogin(Integer.toString(user.hashCode()));
+        user.setPassword(null);
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        }, "Expected RuntimeException");
+    }
+
+    @Test
+    void registrationWithNullLogin_NotOk() {
+        user.setAge(20);
+        user.setLogin(null);
+        user.setPassword(Integer.toString(12345678));
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        }, "Expected RuntimeException");
+    }
+
+    @Test
+    void registrationWithNegativeAge_NotOk() {
+        user.setAge(-15);
+        user.setLogin(Integer.toString(user.hashCode()));
+        user.setPassword(Integer.toString(12345678));
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        }, "Expected RuntimeException");
+    }
```

Comment:
Seems you need to add test for 18 years. It's edge test.

Add tests for all possible User’s parameters (null login/password/age, user under 18/**18**/over 18 years old, negative age, and so on…)